pkg/segment/results/mresults/metricresults.go

```diff
@@ -752,3 +758,48 @@ func (r *MetricsResult) computeAggCount(aggregation structs.Aggregation) {
 	r.DsResults = nil
 	r.State = AGGREGATED
 }
+
+// Retrieve all series values at each timestamp and calculate the results based on those values.
+func (r *MetricsResult) computeAggStdvarOrStddev(aggregation structs.Aggregation) {
+	timestampToVals := make(map[uint32][]float64)
+	r.Results = make(map[string]map[uint32]float64)
+
+	// If we use group by for this vector, we will only perform aggregation for each time series at each timestamp.
+	// Since there is only one value, the standard variance will be 0.
+	if len(aggregation.GroupByFields) > 0 {
```

Comment:
I don't quite understand this comment. I do understand that if we're computing the variance for just one vector, then we compute the variance at each time window and since there's only one point the variance will be 0.

But isn't it possible that we group by something and then each group has more than one series? So then the variance could be non-zero? I think I'm missing something

Reply:
Yes, 1 group may have multiple series, i will modify that